automatic_data_generation/data/base_dataset.py

```diff
@@ -163,12 +167,18 @@ def build_data_files(self, dataset_folder, restrict_to_intent,
             new_train = self.filter_intents(new_train, restrict_to_intent)
             new_test = self.filter_intents(new_test, restrict_to_intent)
 
-        # trim_dataset
+        # trim_dataset : 
         trim_prefix = ''
         if dataset_size is not None:
-            # TODO: stratified shuffle split
             trim_prefix = '_{}'.format(dataset_size)
-            new_train = random.sample(new_train, dataset_size)
+            original_dataset_size = len(new_train)
+            keep_fraction = dataset_size/original_dataset_size
+            from sklearn.model_selection import StratifiedShuffleSplit
+            intents = [row[3] for row in new_train]
```

Comment:
not correct if not SnipsDataset. You could fix that by addind `get_intents` as a child class method.